support-workers/src/main/scala/com/gu/support/workers/lambdas/CreateZuoraSubscription.scala

```diff
@@ -81,6 +81,7 @@ object CreateZuoraSubscription {
         case None => subscribe(subscribeItem, zuoraService).map(response => toHandlerResult(state, response, paymentMethodWithPaymentSchedule, requestInfo))
           .withLogging("subscribe")
       }
+      _ <- updateRedemptionCode(state.paymentMethod, subscriptionData, SetCodeStatus.withDynamoLookup(redemptionService))
```

Comment:
I realise this wasn't introduced in this PR but maybe move the match against `maybeDomainSubscription` above to a separate method? This for comprehension is getting pretty big

Reply:
good idea.  the whole skip/don't skip thing seems to involve several methods that all call each other dotted around, but for simplicity I had decided that was out of scope for this PR or a preceding refactor.  Happy to add a health card to revisit in a separate PR though?